Sources/DatadogCrashReporting/PLCrashReporterIntegration/DDCrashReportExporter.swift

```diff
@@ -0,0 +1,190 @@
+/// are formatted using Apple-like format. The implementation is based on the PLCR's formatter, ref.:
+/// https://github.com/microsoft/plcrashreporter/blob/master/Source/PLCrashReportTextFormatter.m
+internal struct DDCrashReportExporter {
+    private let unknown = "<unknown>"
+    private let unavailable = "???"
+
+    func export(_ crashReport: CrashReport) -> DDCrashReport {
+        return DDCrashReport(
+            date: crashReport.systemInfo?.timestamp,
+            type: formatType(for: crashReport),
+            message: formatMessage(for: crashReport),
+            stack: formatStack(for: crashReport),
+            threads: threads(from: crashReport),
+            binaryImages: binaryImages(from: crashReport),
+            meta: meta(for: crashReport),
+            wasTruncated: false, // TODO: RUMM-1462 Send flag telling if any stack trace was truncated
+            context: crashReport.contextData
+        )
+    }
+
+    // MARK: - Formatting `error.type`, `error.message` and `error.stack`
+
+    /// Formats the error type - in Datadog Error Tracking this corresponds to `error.type`.
+    ///
+    /// **Note:** This value is used for building error's fingerprint in Error Tracking, thus its cardinality must be controlled.
+    private func formatType(for crashReport: CrashReport) -> String {
+        return "\(crashReport.signalInfo?.name ?? unknown) (\(crashReport.signalInfo?.code ?? unknown))"
+    }
+
+    /// Formats the error message - in Datadog Error Tracking this corresponds to `error.message`.
+    ///
+    /// **Note:** This value is used for building error's fingerprint in Error Tracking, thus its cardinality must be controlled.
+    private func formatMessage(for crashReport: CrashReport) -> String {
+        if let exception = crashReport.exceptionInfo {
+            // If the crash was caused by an uncaught exception
+            let exceptionName = exception.name ?? unknown // e.g. `NSInvalidArgumentException`
+            let exceptionReason = exception.reason ?? unknown // e.g. `-[NSObject objectForKey:]: unrecognized selector sent to instance 0x...`
+            return "Terminating app due to uncaught exception '\(exceptionName)', reason: '\(exceptionReason)'."
+        } else {
+            // Use signal description available in OS
+            guard let signalName = crashReport.signalInfo?.name else { // e.g. SIGILL
+                return "Application crash: \(unknown)"
+            }
+
+            let knownSignalNames = Mirror(reflecting: sys_signame)
+                .children
+                .compactMap { $0.value as? UnsafePointer<Int8> }
+                .map { String(cString: $0).uppercased() } // [HUP, INT, QUIT, ILL, TRAP, ABRT, ...]
+
+            let knownSignalDescriptions = Mirror(reflecting: sys_siglist)
+                .children
+                .compactMap { $0.value as? UnsafePointer<Int8> }
+                .map { String(cString: $0) } // [Hangup, Interrupt, Quit, Illegal instruction, ...]
```

Comment:
Shouldn't you keep this in a static or class dictionary  instead of computing them all the time ? Are they depending in any way on this method arguments or am I missing something ?

Reply:
Good idea 👍. We can consider these values constant as it's hard to imagine that this will change in the OS internals. I changed to use static dictionary in implementation and added OS-value inference in tests (this way we will spot that it changed eventually). Done in https://github.com/DataDog/dd-sdk-ios/pull/536/commits/2d68a62623d101c4ae75f83bd8ad16e5aca64590.